tests/unit/model/relationships/rollback-test.js

```diff
@@ -0,0 +1,275 @@
+    });
+
+    env = setupStore({ person: Person, dog: Dog });
+    store = env.store;
+  }
+});
+
+test("saved changes to relationships should not roll back to a pre-saved state (from child)", function(assert) {
+  var person1, person2, dog1, dog2, dog3;
+
+  env.adapter.updateRecord = function(store, type, snapshot) {
+    return Ember.RSVP.resolve({ id: 2, owner: 1 });
+  };
+
+  run(function() {
+    store.push({
+      data: {
+        type: 'person',
+        id: 1,
+        attributes: {
+          firstName: "Tom",
+          lastName: "Dale"
+        },
+        relationships: {
+          dogs: {
+            data: [{
+              type: 'dog',
+              id: 1
+            }]
+          }
+        }
+      }
+    });
+    store.push({
+      data: {
+        type: 'person',
+        id: 2,
+        attributes: {
+          firstName: "John",
+          lastName: "Doe"
+        },
+        relationships: {
+          dogs: {
+            data: [{
+              type: 'dog',
+              id: 2
+            }]
+          }
+        }
+      }
+    });
+    store.push({
+      data: {
+        type: 'dog',
+        id: 1,
+        attributes: {
+          name: "Fido"
+        },
+        relationships: {
+          owner: {
+            data: {
+              type: 'person',
+              id: 1
+            }
+          }
+        }
+      }
+    });
+    store.push({
+      data: {
+        type: 'dog',
+        id: 2,
+        attributes: {
+          name: "Bear"
+        },
+        relationships: {
+          owner: {
+            data: {
+              type: 'person',
+              id: 2
+            }
+          }
+        }
+      }
+    });
+    store.push({
+      data: {
+        type: 'dog',
+        id: 3,
+        attributes: {
+          name: "Spot"
+        },
+        relationships: {
+          owner: {
+            data: null
+          }
+        }
+      }
+    });
+    person1 = store.peekRecord('person', 1);
+    person2 = store.peekRecord('person', 2);
+    dog1 = store.peekRecord('dog', 1);
+    dog2 = store.peekRecord('dog', 2);
+    dog3 = store.peekRecord('dog', 3);
+
+    person1.get('dogs').addObject(dog2);
+  });
+
+  run(function() {
+    dog2.save().then(function () {
+      person1.get('dogs').addObject(dog3);
+      return Ember.RSVP.all([dog2.rollback(), dog3.rollback()]);
+    }).then(function () {
+      person1.get('dogs').then(function (dogs) {
+        assert.deepEqual(dogs.toArray(), [dog1,dog2]);
+      });
+      person2.get('dogs').then(function (dogs) {
+        assert.deepEqual(dogs.toArray(), []);
+      });
+      dog1.get('owner').then(function (owner) {
+        assert.equal(owner, person1);
+      });
+      dog2.get('owner').then(function (owner) {
+        assert.equal(owner, person1);
+      });
+    });
+  });
+});
+
+// skip("saved changes to relationships should not roll back to a pre-saved state (from parent)", function(assert) {
```

Comment:
So after thinking about this more, it is probably more relevant to restart the discussion on https://github.com/emberjs/rfcs/pull/21 about the above concerns. What could be better than specifying options in `rollback()` might be the suggested approach in the rfc along the lines of `posts: hasMany('post', { dirties: 'relationship' }),` etc and rollback just reading from that initial config instead.